sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/models/ListFilesAndDirectoriesSegmentResponse.java

```diff
@@ -34,14 +34,14 @@ public final class ListFilesAndDirectoriesSegmentResponse {
     /*
      * The DirectoryPath property.
      */
-    @JacksonXmlProperty(localName = "DirectoryPath", isAttribute = true)
-    private String directoryPath;
+    @JsonProperty(value = "DirectoryPath", required = true)
```

Comment:
Did the service change how this is being returned? This will break backwards compatibility with older service versions on the SDK side

Reply:
The service added an optional XML attribute to the several fields that were previous just strings.  This attribute is only returned if the string content is encoded - [spec](https://microsoft.sharepoint.com/:w:/t/AzureStorage/ET5IdTnpIqdFhaMKHD3S_OQBHdC62LQMQh393G9nxLXf3Q).

Reply:
Yeah, got that, that these fields now have a new XML attribute to indicate if the value is encoded but it looks to me that either the Swagger update is wrong or something is wrong in Java codegen. Based on the document there is a new attribute along side the `DirectoryPath` XML attribute but this code generation now thinks this property shows up as `<DirectoryPath Encoded="true">directory-path</DirectoryPath>` instead of `<EnumerationResults DirectoryPath="directory-path" Encoded="true">...`

Reply:
We fixed this issue in swagger.